### PR TITLE
Manual task exe

### LINKED ISSUE #465: Execute planned task immediately
Sometimes a task for relocation or move in exists but the member gets into the room early.
It would be nice to have a button to execute a task immediately instead of deleting the task and performing the action manually.

### PR DESCRIPTION
**Pull request by lukasjuhrich**
- Fixes #465
- Also adds a unit test for `cancel_task`

Screenshots: 
image
image

### CODE REVIEW

File: pycroft/lib/task.py [hunk 1]
```diff
@@ -192,10 +198,22 @@ def get_active_tasks_by_type(type):
...
     log_task_event("Cancelled task {}.".format(task.id), processor, task)
 
     task.status = TaskStatus.CANCELLED
+
+
+def manually_execute_task(task: Task, processor: User):
+    if task.status != TaskStatus.OPEN:
+        raise ValueError("Cannot execute a task that is not open")
+
+    get_task_implementation(task).execute(task)
```
Comment (ibot3) [inline pycroft/lib/task.py:214]: Catching errors here could be a good idea, because when an exception occurs, the user only sees a 500 error message.

**Review by ibot3 — APPROVED** [review]

File: pycroft/lib/task.py (hunk 1 above)
Comment (lukasjuhrich) [reply pycroft/lib/task.py:214]: The question would be which errors to catch. A `except Exception:` would not be advisable. We want to have programming errors be fatal and reported.
However, if it's an error of the form „does not live in that room“ or sth, we will probably want to catch that.

But what does that mean to thhe lifecycle of the task? Does it go to `TaskState.FAILED`?

Comment (ibot3) [reply pycroft/lib/task.py:214]: I don't think that it is an good idea to reimplement the task execution logic here again. So we should either

- Do not change the state to `FAILED` but just display an error message instead

or

- Don't execute the task here directly, but instead trigger that the task should be executed by celery. For example by setting the due date to the current timestamp and calling `pycroft.task.execute_scheduled_tasks.delay()`.

Comment (lukasjuhrich) [reply pycroft/lib/task.py:214]: I think this is a good time to introduce a `PycroftException` for `lib` errors.

Comment (lukasjuhrich) [reply pycroft/lib/task.py:214]: See #493

Comment (lukasjuhrich) [reply pycroft/lib/task.py:214]: Resolved in f34fea5261b30ba55a6e5c5a5a00775a195427b3.

**Pull request merged**